### PR TITLE
labeler.yml: Don't apply CI-all-test for west.yml

### PR DESCRIPTION
**Pull request by oyvindronningstad**
The author should instead apply the right labels, since most updates
don't concern the whole tree.

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Review by thst-nordic — APPROVED** [review]

**Comment by ioannisg** [comment]
I think that's probably OK, but it should come with some documentation in the contribution guidelines, stressing what the author needs to do.

**Review by tejlmand — CHANGES_REQUESTED** [review]
> The author should instead apply the right labels, since most updates
don't concern the whole tree.

I strongly disagree.
Today we have this principle in sdk-nrf, and people forget to add  the right testing labels there.
For example they make changes to Bluetooth code, and forget to apply `ci-ble-test`.

If they cannot do this in sdk-nrf, I disapprove having the same for `west.yml`.

People can still remove the `ci-all-test` label and then add specific labels for the tests they actually need.need.

**Comment by tejlmand** [comment]
As code owners we already spend a lot of time on investigating CI issues during release work, and i'm afraid this will become even worse if manifest PRs are not tested per default as more PRs may be merged that actually breaks CI, but that will not be detected until someone having a PR with that test label, and thus we'll need to spend even more time chasing down from where that error originated.

And the PR that failed will be blocked, even though that PR had nothing to do with the failure.

**Comment by lemrey** [comment]
It's just taxing for authors to remove and reapply the appropriate labels every time they push a commit (after waiting for the bot to trigger). This can happen easily several times for each PR, and it would be much less work to let the author apply the proper labels, once and for all, and let the code-owner make sure of that before the PR is merged.

In the end, the code-owners are the one who merge PRs and they should accept the responsibility to make sure the correct labels are applied.

**Comment by tejlmand** [comment]
> In the end, the code-owners are the one who merge PRs and they should accept the responsibility to make sure the correct labels are applied.

No, that's the responsibility of the author, unless code owners should start to only merge PRs where they have domain knowledge.
But in that case, you might need to wait longer time before getting things merged.

As I said, as long as people don't apply labels at `sdk-nrf` then I don't like to extend that scheme to manifest updates.
For example, why was this not testing anything:
https://github.com/nrfconnect/sdk-nrf/pull/4787

**Comment by tejlmand** [comment]
> It's just taxing for authors to remove and reapply the appropriate labels every time they push a commit (after waiting for the bot to trigger)

Now, that would be a matter of asking the action to only apply the labels when the PR is opened.
Which is very different from what this PR suggests.

**Comment by lemrey** [comment]
Having the bot trigger once instead of every push would be good enough for me.

**Comment by oyvindronningstad** [comment]
Question, since the labeler and CI are triggered at the same time, I assume you only have a short window of time to change the labels before CI reads them, how long is that window. @thst-nordic do you know?

**Comment by thst-nordic** [comment]
~20min   
this is how long it takes to run all the twister targets and build all the samples/apps.

**Comment by oyvindronningstad** [comment]
Idea: Can we make https://github.com/zephyrproject-rtos/action-manifest apply the right CI labels? It already parses the manifest for the changed modules, and already applies labels (DNM++). @carlescufi @thst-nordic?

**Comment by oyvindronningstad** [comment]
> 
> 
> Idea: Can we make https://github.com/zephyrproject-rtos/action-manifest apply the right CI labels? It already parses the manifest for the changed modules, and already applies labels (DNM++). @carlescufi @thst-nordic?

AFAICT, we would need to

1. Add a label-suffix, analogous to label-prefix
2. Change the relevant CI labes to match the module names from the manifests. If we also change the labels to remove "-test", we don't need 1.

cc @mbolivar-nordic

**Comment by mbolivar-nordic** [comment]
@oyvindronningstad 

> Idea: Can we make https://github.com/zephyrproject-rtos/action-manifest apply the right CI labels?

Probably

> AFAICT, we would need to

I'm sorry, I'm having trouble understanding your shorthands. I looked at all the existing labels and I'm not sure how to apply your suggestion to them though.

Could you give an example, e.g. of what you want to happen when someone changes the find my repository in west.yml?

I assume we also want to run tests on everything if zephyr changes.

**Comment by oyvindronningstad** [comment]
@mbolivar-nordic

>Could you give an example, e.g. of what you want to happen when someone changes the find my repository in west.yml?

It applies the CI-find-my-test label.

>I assume we also want to run tests on everything if zephyr changes.

I agree

I created a quick PR that does something like this:
https://github.com/zephyrproject-rtos/action-manifest/pull/2
nrf: https://github.com/nrfconnect/sdk-nrf/pull/4817

**Comment by mbolivar-nordic** [comment]
> I created a quick PR that does something like this:

The approach seems sound to me.

**Comment by oyvindronningstad** [comment]
Superseded by #4817

**Pull request closed without merging**
